kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/MetricsST.java

```diff
@@ -128,8 +128,9 @@ void cleanUp() {
     @SuppressWarnings("java:S2925")
     @BeforeEach
     void beforeEach(String namespace) throws InterruptedException {
-        final String scraperName = namespace + "-" + Constants.SCRAPER_LABEL_VALUE;
-        resourceManager.createResourceFromBuilderWithWait(ScraperTemplates.scraperPod(namespace, scraperName));
+        final String scraperName = Constants.SCRAPER_LABEL_VALUE;
+
+        ScraperTemplates.deployPortIdentifiesNodeWithNoFilters(namespace, scraperName);
```

Comment:
If I understand this right, this is a deploying another proxy whose only purpose is to be the pod that you exec into in order to run curl commands to collect the metrics from the real proxy.  That really violates least surprise that'll confuse developers who won't understand why there are two proxies seemingly running.  At very least you need to document what's going on.  Better would be to run curl command from a dedicated container (perhaps using an image from https://hub.docker.com/r/curlimages/curl).

Reply:
Oops, I missed that this ended up with two proxies running. I agree thats confusing, I'd be ok deploying a customised instance that we could scrape but deploying a new instance to scrape another isn't helpful.

Reply:
Yeah you are right. I was blind to that part and wanted to remove as much as possible. Sorry for that, but I'll leave it as it was as it does not make any sense at all.